packages/composite-checkout-wpcom/src/components/wp-order-review-line-items.js

```diff
@@ -84,6 +84,9 @@ function WPLineItem( {
 						isVisible={ isModalVisible }
 						closeModal={ () => {
 							setIsModalVisible( false );
+							onEvent( {
```

Comment:
While this will generate an event, I'm pretty sure that `closeModal` is called any time that the modal is closed, for both success and cancel. I think you'll need to modify `CheckoutModal` so it differentiates between primaryAction close and the other versions of close.

Reply:
🤦‍♂ good catch.

Reply:
@sirbrillig: updated - this should be good for review.